workflows/flows/analysis/assembly/tasks/make_samplesheet_assembly.py

```diff
@@ -0,0 +1,205 @@
+                    "http://"
+                    + ftp_path
+                ),
+            ),
+        },
+        bludgeon=True,
+    )
+
+    logger.info(f"Created assembly samplesheet at {sample_sheet_csv}")
+    return sample_sheet_csv, ss_hash
+
+
+@task
+def make_samplesheet_for_map(
+    assembly_analysis_id: uuid.UUID,
+    output_dir: Path = None,
+) -> (Path, str):
+    """
+    Makes a samplesheet for a batch of assembly analyses for MAP.
+
+    Uses filesystem structure instead of database queries to find required files.
+    Assumes results follow the expected pipeline structure:
+    - ASA CDS GFF: {asa_workspace}/{assembly_accession}/cds/{assembly_accession}_predicted_cds.gff.gz
+    - VIRify GFF: {virify_workspace}/{assembly_accession}/08-final/gff/{assembly_accession}_virify.gff.gz
+
+    The samplesheet has the following columns:
+    - sample: the assembly first_accession
+    - assembly: the assembly fasta (from ENA metadata)
+    - user_proteins_gff: the assembly analysis CDS GFF file
+    - virify_gff: the VIRify GFF file
+
+    :param assembly_analysis_id: The AssemblyAnalysisBatch id containing analyses to process
+    :param output_dir: Directory where the samplesheet should be saved. If None, uses default workdir.
+    :return: Tuple of the Path to the samplesheet file, and a hash of the assembly IDs which is used in the SS filename.
+    """
+    logger = get_run_logger()
+
+    batch = AssemblyAnalysisBatch.objects.get(id=assembly_analysis_id)
+
+    mgnify_study = batch.study
+
+    # We only keep those that are VIRify completed (which requires them to be ASA completed)
+    completed_analysis_ids = batch.batch_analyses.filter(
+        virify_status=AssemblyAnalysisPipelineStatus.COMPLETED
+    ).values_list("analysis_id", flat=True)
+
+    assembly_analyses = batch.analyses.filter(
+        id__in=completed_analysis_ids
+    ).select_related("assembly")
+
+    assemblies = analyses.models.Assembly.objects.filter(analyses__in=assembly_analyses)
+
+    logger.info("Making MAP samplesheet for assemblies.")
+
+    ss_hash = queryset_hash(assemblies, "id")
+
+    # Use provided output_dir or fall back to the default workdir
+    if output_dir is None:
+        output_dir = Path(EMG_CONFIG.slurm.default_workdir)
+    else:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    # Get pipeline workspaces from batch
+    asa_workspace = batch.get_pipeline_workspace(AssemblyAnalysisPipeline.ASA)
+    virify_workspace = batch.get_pipeline_workspace(AssemblyAnalysisPipeline.VIRIFY)
+
+    # Create a filename for the MAP samplesheet
+    map_samplesheet_filename = (
+        output_dir / f"{mgnify_study.ena_study.accession}_samplesheet_map_{ss_hash}.csv"
+    )
+
+    # Create the samplesheet with the required columns
+    with open(map_samplesheet_filename, "w", newline="") as csvfile:
+        fieldnames = ["sample", "assembly", "user_proteins_gff", "virify_gff"]
```

Comment:
what does "user" refer to?

Reply:
```
When user_proteins_gff is provided, three more GFF files will be generated:

    user_mobilome_clean.gff: mobilome + its associated CDSs
    user_mobilome_extra.gff: mobilome + ViPhOGs annotated genes (note that ViPhOG annotation is generated by VIRify)
    user_mobilome_full.gff: mobilome + any other annotation on user gff
```
from - https://github.com/EBI-Metagenomics/mobilome-annotation-pipeline

Probably not the best name, we can look into chaning that later.